### PR TITLE
iOS 26 compatibility (fix SwiftData issues)

### LINKED ISSUE #81: ModelContext Creation Crash on iOS 26
### Description

Spezi Scheduler fails when initializing the `ModelContext`:
https://github.com/StanfordSpezi/SpeziScheduler/blob/af631078adbb0b18925256af406ccba4fe7f6745/Sources/SpeziSchedulerUI/TestingSupport/SchedulerSampleData.swift#L49

The debug console logs:
```console
Loading module(s) Scheduler, SchedulerNotifications, TestAppScheduler ...
SwiftData/SchemaProperty.swift:286: Fatal error: Illegal attempt to create a property that's a sequence of a non-codable type (arguments - Array<FormatArgument>). Did you mean to use a transformable attribute?
```

### Reproduction

Launch the Testing App in the repo on an iOS 26 simulator.

### Expected behavior

Spezi Scheduler should load correctly and any schema changes should be backward compatible.

### Additional context

It seems like `FormatArgument` is located in `SwiftUI.LocalizedStringKey.FormatArgument`. It might make sense to move away from the localized strings and rather store the plain values? Or some other way to better derive the localized value at runtime?

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct and Contributing Guidelines

### PR DESCRIPTION
**Pull request by lukaskollmer**
# iOS 26 compatibility (fix SwiftData issues)

## :recycle: Current situation & Problem
this PR fixes #81, by moving away from directly having attributes of type `String.LocalizationValue` (`Task.title` and `Task.instructions`), and gracefully migrating over existing values if applicable.

the underlying issue here is the fact that iOS 26 seems to make some changes in the internal representation of `String.LocalizedValue`, which cause the type (despite being `Codable`) to no longer be SwiftData-compatible, leading to a crash in the schema creation.

while we can fairly easily work around no longer being able to store the `LocalizationValue` directly (we now store it as a plist-encoded Data, akin to how the `Schedule.recurrenceRule` property is persisted), migrating over existing values for these attributes is sadly less elegant.
we cannot use a normal SwiftData migration (it requires the old and new schema be present, but creating the schema is the exact place where our crash is); instead what we do is that we perform a one-time custom migration which consists of the following steps:
1. check if we need to perform our custom iOS 26 migration
2. (assuming yes) directly access the sqlite database and extract, for every row in the `Task` type's table, the columns corresponding to the title and instructions properties
    - these are actually multiple columns per property, since the `String.LocalizationValue` doesn't simply get JSON-encoded and put into a single BLOB column; instead it essentially gets "embedded" directly into the table, and there is one column for each of the `String.LocalizationValue`'s 2 internal properties: `key` and `arguments`
    - since `Task` has 2 properties of type `String.LocalizationValue`, we end up with 4 columns we need to fetch the values from: `ZKEY` and `ZARGUMENTS` (which correspond to the `instructions` property), and `ZKEY1` and `ZARGUMENTS1` (which correspond to the `title` property).
    - curiously enough, the embedding of struct properties directly into the table schema only seems to happen at one level; the `String.LocalizationValue`'s `arguments` property simply gets stored as JSON-encoded BLOB.
3. keep track of all task's titles and instructions using a small data structure
4. normally open the ModelContainer using SwiftData; this will perform an autimatic lighweight migration that removes the `title` and `instructions` properties, and adds the new `titleData` and `instructionsData` properties (both of which will be nil-initialized)
5. apply the task titles and instructions we've extracted to the ModelContext; this will effectively restore the values the properties had before
6. set a small flag to keep track of the fact that we've already performed this migration and should skip it going forward.

we also use this PR as an opportunity to move the scheduler's database from directly in the application's `~/Documents` directory to a special `~/Documents/SpeziScheduler` directory.

## :gear: Release Notes
- iOS 26 compatibility

## :books: Documentation
doesn't affect the public API but i'll add some internal docs to explain what's going on and why

## :white_check_mark: Testing
i added a test that opens an old model context and checks that everything gets migrated properly

## :pencil: Code of Conduct & Contributing Guidelines
By creating and submitting this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by PSchmiedmayer — APPROVED** [review]
Very cool approach; thank you for taking a look at this @lukaskollmer 🚀

Would like to get some additional eyes on this from @bauer-andreas, but I think it looks solid. A bit unfortunate that this is needed but seems like these properties are the only remaining ones where we don't use Data representations for more complex types so we might be well set for the future.

**Comment by lukaskollmer** [comment]
@PSchmiedmayer tbh i'm not overly happy about introducing a new dependency here, but it seems like the package is pretty lightweight overall; their only dependency is apple's [swift-toolchain-sqlite](https://github.com/swiftlang/swift-toolchain-sqlite) package. we could also directly interface with the sqlite C api that's shipped with iOS, but that'd probably make our code here significantly more complex (and error-prone)

**Comment by PSchmiedmayer** [comment]
@lukaskollmer Agree that the dependency is not ideal, but I think we should be fine with it for now. It will only be used in a few edge cases and I can imagine that it might be useful in the future as well.

File: .github/workflows/build-and-test.yml
Comment (PSchmiedmayer) [inline .github/workflows/build-and-test.yml:1]: Thank you for updating the CI setup here!

File: Tests/SpeziSchedulerTests/SchedulerTests.swift [hunk 1]
```diff
@@ -818,6 +818,49 @@ struct SchedulerTests { // swiftlint:disable:this type_body_length
             #expect(event.occurrence.start == expectedDate)
         }
     }
+    
+    @Test
+    func iOS26Migration() throws {
```
Comment (PSchmiedmayer) [inline Tests/SpeziSchedulerTests/SchedulerTests.swift:823]: Thank you for all the tests here; amazing!!

**Review by PSchmiedmayer — APPROVED** [review]
Happy to see this merged, thank you for all the work here @lukaskollmer, great effort for the team!! 🚀
